### PR TITLE
Rename StoreMixin to Mixin; Support React context

### PR DESCRIPTION
**Pull request by mridgway**
* **BC Break** `require('fluxible').StoreMixin` is changed to `require('fluxible').Mixin`

@lingyan

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: index.js [hunk 1]
```diff
@@ -4,4 +4,4 @@
  */
 module.exports = require('./lib/FluxibleApp');
 module.exports.FluxibleContext = require('./lib/FluxibleContext');
-module.exports.StoreMixin = require('./mixins/StoreMixin');
+module.exports.Mixin = require('./mixins/StoreMixin');
```
Comment (redonkulus) [inline index.js]: Should be FluxibleMixin

File: mixins/FluxibleMixin.js [hunk 2]
```diff
@@ -34,10 +34,14 @@
  *     ...
  * });
  */
-var DEFAULT_CHANGE_HANDLER = 'onChange',
-    StoreMixin;
+var DEFAULT_CHANGE_HANDLER = 'onChange';
+var React = require('react');
 
-StoreMixin = {
+var StoreMixin = {
```
Comment (lingyan) [inline mixins/FluxibleMixin.js]: rename to FluxibleMixin

File: mixins/FluxibleMixin.js [hunk 3]
```diff
@@ -53,6 +57,18 @@ StoreMixin = {
     },
 
     /**
+     * Calls an action
+     * @method executeAction
+     */
+    executeAction: function executeAction() {
+        var context = this.props.context || this.context;
+        if (!context || !context.executeAction) {
+            throw new Error('executeAction was called but no context was provided');
```
Comment (lingyan) [inline mixins/FluxibleMixin.js]: make the error message more helpful to users by telling them we expect it from this.props.context or this.context.

File: mixins/FluxibleMixin.js [hunk 4]
```diff
@@ -61,10 +77,11 @@ StoreMixin = {
     getStore: function (store) {
         var storeInstance = store;
         if ('object' !== typeof storeInstance) {
-            if (!this.props.context) {
+            var context = this.props.context || this.context;
+            if (!context) {
```
Comment (lingyan) [inline mixins/FluxibleMixin.js]: should we check for context.getStore as well?

File: tests/fixtures/applications/basic/components/Timestamp.jsx [hunk 5]
```diff
@@ -6,8 +6,10 @@
 
 var React = require('react/addons'),
     updateTime = require('../actions/updateTime');
+var FluxibleMixin = require('../../../../../mixins/FluxibleMixin').Mixin;
```
Comment (lingyan) [inline tests/fixtures/applications/basic/components/Timestamp.jsx]: is `.Mixin` needed?

**Comment by redonkulus** [comment]
This looks good, just @lingyan comment left.

**Comment by jedireza** [comment]
Woo hoo! :+1:

**Comment by redonkulus** [comment]
:shipit:

**Comment by lingyan** [comment]
:ship:

**Comment by mridgway** [comment]
Releasing as minor version bump.

**Comment by gpbl** [comment]
@mridgway why not calling the export `FluxibleMixin`? In ES6 now I write:

```
import { Mixin as FluxibleMixin } from 'fluxible'
```

instead I could write:

```
import { FluxibleMixin } from 'fluxible'
```

**Comment by mridgway** [comment]
@gpbl Noted. Maybe we can add an alias for now.

**Comment by gpbl** [comment]
Thanks!

**Pull request merged**